### PR TITLE
DEV-3521: Fix scheduled payment email header

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

- Fixes alignment issue with scheduled payment reminder emails
- Adds a debug-only route to the Django app, `/__debug_email__/recurring-contribution/` that allow previewing emails.

#### Why are we doing this? How does it help us?

- Fixes a display bug
- Makes it easier to work on styled emails

#### How should this be manually tested? Please include detailed step-by-step instructions.

To test styled emails for regressions:

1. Set up a recurring contribution to an org whose default page has a style set.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (header color and logo are correct, appear above text header) and contains accurate info.
4. Set up a recurring contribution to an org whose default page has a style set, but no header logo.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (has left-aligned NRE logo at top, no background color in header) and contains accurate info.

To test unstyled emails for regressions:

1. Set up a recurring contribution to an org whose default page doesn't have a style set.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (has left-aligned NRE logo at top) and contains accurate info.

To test free plan emails for regressions:

1. Set up a recurring contribution to an org whose default page has a style set, but whose org is on the free plan.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (does not use custom style, has left-aligned NRE logo at top) and contains accurate info.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

I updated existing Python tests.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

 I also added notes to the readme explaining how the debug email routes work.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3521

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: apps/emails/templates/recurring-contribution-email-reminder.html [hunk 1]
```diff
@@ -31,15 +34,17 @@
 {% endblock custom_style %}
 
 {% block header %}
+{% if style.logo_url %}
 <tr>
-  {% if style.logo_url %}
-  <td class="header-background" align="center" valign="top">
-    <img class="header-logo" src="{{ style.logo_url }}" alt="logo" >
+  <td class="header-background">
+    <img class="header-logo" src="{{ style.logo_url }}" alt="" />
```
Comment (GuiMend) [inline apps/emails/templates/recurring-contribution-email-reminder.html:40]: alt information was removed by accident?

File: apps/emails/templates/recurring-contribution-email-reminder.html [hunk 2]
```diff
@@ -31,15 +34,17 @@
...
   </td>
-  {% endif %}
+</tr>
+{% endif %}
+<tr>
```
Comment (GuiMend) [inline apps/emails/templates/recurring-contribution-email-reminder.html:44]: Is there a reason you opted for having the logo and header in separate `<tr>` ?

Comment (nrh-cklimas) [reply apps/emails/templates/recurring-contribution-email-reminder.html:44]: They're meant to vertically stack in the most recent Figma mocks. I'll link in Jira.

**Review by GuiMend — APPROVED** [review]

**Comment by nrh-cklimas** [comment]
I just merged DEV-3519 into this branch, so there are additional changes here now.

**Review by x110dc — APPROVED** [review]
This LGTM, but I'd like @njharman to review it too.

File: README.md [hunk 3]
```diff
@@ -420,6 +421,21 @@ If you have a need to run or test tasks using a Celery worker, there are some Ma
...
 
+## Email Template Development
+
+If `DEBUG` is set in Django settings, then the app will serve example emails under `/__debug_emails__/`. This can be useful for testing template changes.
+
+Routes that currently exist:
+
+- `http://localhost:8000/__debug_emails__/contribution-confirmation/` shows a
+  contribution receipt email. A revenue program ID set in the query string as
+  `rp` is required, e.g. `/recurring-contribution/?rp=3`. You can also override
```
Comment (njharman) [inline README.md:432]: minor possible typo, this line is `recurring-...` but talking about `contribution-...`

Both use same parameter.  I would have included `?rp=3` in url on line 430 and cut `e.g. ...` .  For copy paste ease, and for people like me who are just gonna grab that URL before reading the comment, have it not work, then go back to read what they did wrong and feel bad for not reading your comment in first place.

**Review by njharman — APPROVED** [review]
Looks good. Just reviewed Python code, assuming QA testing will verify that the css/html does what AC says it should.

One minor possible typo in README. No need for rereview if indeed typo and fixed.

File: README.md (hunk 3 above)
Comment (nrh-cklimas) [reply README.md]: :+1: 69c70752884c440d119dc902d4a07c5ccdc9525a

**Pull request merged**